src/tests/ftest/util/server_utils.py

```diff
@@ -50,6 +58,422 @@ class ServerFailed(Exception):
+            """ Create a DaosServerConfig object."""
+            super(DaosServer.DaosServerConfig, self).__init__(
+                "/run/server_config/*")
+            self.data = None
+            self.name = BasicParameter(None, "daos_server")
+            self.port = BasicParameter(None, "10001")
+            self.nvme = None
+            self.scm = None
+
+        @property
+        def name(self):
+            """Get the name from the server config."""
+            if self.data and "name" in self.data:
+                return self.data["name"]
+            return None
+
+        @name.setter
+        def name(self, value):
+            """Set the server config name attribute."""
+            if self.data and "name" in self.data:
+                self.data["name"] = value
+
+        @property
+        def port(self):
+            """Get the port from the server config."""
+            if self.data and "port" in self.data:
+                return self.data["port"]
+            return None
+
+        @port.setter
+        def port(self, value):
+            """Set the port config attribute."""
+            if self.data and "port" in self.data:
+                self.data["port"] = value
+
+        def get_params(self, test):
+            """Get values for all of the command params from the yaml file.
+
+            If no key matches are found in the yaml file the BasicParameter
+            object will be set to its default value.
+
+            Args:
+                test (Test): avocado Test object
+            """
+            super(DaosServer.DaosServerConfig, self). get_params(test)
+            # Read the baseline conf file data/daos_server_baseline.yml
+            try:
+                with open('{}/{}'.format(test.basepath, DEFAULT_FILE), 'r') \
+                as rfile:
+                    self.data = yaml.safe_load(rfile)
+            except Exception as err:
+                self.log.info("<SERVER> Exception occurred: %s" % str(err))
+                traceback.print_exception(
+                    err.__class__, err, sys.exc_info()[2])
+                raise ServerFailed(
+                    "Failed to Read {}/{}".format(test.basepath, DEFAULT_FILE))
+
+            # Read the values from avocado_testcase.yaml file if test ran
+            # with Avocado.
+            new_value_set = {}
+            if "AVOCADO_TEST_DATADIR" in os.environ:
+                avocado_yaml_file = str(
+                    os.environ["AVOCADO_TEST_DATADIR"]).split(".")[0] + ".yaml"
+
+                # Read avocado test yaml file.
+                try:
+                    with open(avocado_yaml_file, 'r') as rfile:
+                        filedata = rfile.read()
+                    # Remove !mux for yaml load
+                    new_value_set = yaml.safe_load(
+                        filedata.replace('!mux', ''))
+                except Exception as err:
+                    self.log.info("<SERVER> Exception occurred: %s" % str(err))
```

Comment:
(pylint-logging-not-lazy) Specify string format arguments as logging function parameters